src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java

```diff
@@ -1,16 +1,46 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.DataProcessingException;
 import mate.academy.hibernate.relations.model.Actor;
 import mate.academy.hibernate.relations.service.ActorService;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
 
 public class ActorServiceImpl implements ActorService {
+    private final SessionFactory factory;
+
+    public ActorServiceImpl(SessionFactory sessionFactory) {
+        this.factory = sessionFactory;
+    }
+
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
+            transaction.commit();
+            return actor;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Error adding Movie: " + actor, e);
```

Comment:
The error message in the `DataProcessingException` should refer to 'Actor' instead of 'Movie'. This seems to be a copy-paste error.